### PR TITLE
Add resource for global init scripts

### LINKED ISSUE #295: [FEATURE] Support for new Global Init Scripts
New Global Init Scripts are now in public preview.  [Global Init Scripts API](https://docs.databricks.com/dev-tools/api/latest/global-init-scripts.html)

##### This implementation may look like:
```hcl
resource "databricks_global_init_script" "script" {
  script_content = base64encode("./script/path")
  name = "init_script"
  position = 1
  enabled = true
}
```

### PR DESCRIPTION
**Pull request by alexott**
This fixes #295

### CODE REVIEW

File: compute/resource_global_init_script.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+				ContentBase64: base64.StdEncoding.EncodeToString(content),
+				Enabled:       d.Get("enabled").(bool),
+				Position:      int32(d.Get("position").(int)),
+				Name:          d.Get("name").(string),
+			})
+			if err != nil {
+				return err
+			}
+			d.SetId(scriptID)
+			scriptStatus, err := globalInitScriptsAPI.Get(scriptID)
```
Comment (nfx) [inline compute/resource_global_init_script.go:72]: `.Get` is already called by `Read` method (handled in `CommonResource`). so lines 72-76 are not needed

File: compute/resource_global_init_script.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			if err = globalInitScriptsAPI.Update(scriptID, GlobalInitScriptPayload{
+				ContentBase64: base64.StdEncoding.EncodeToString(content),
+				Enabled:       d.Get("enabled").(bool),
+				Position:      int32(d.Get("position").(int)),
+				Name:          d.Get("name").(string),
+			}); err != nil {
+				return err
+			}
+
+			scriptStatus, err := globalInitScriptsAPI.Get(scriptID)
```
Comment (nfx) [inline compute/resource_global_init_script.go:107]: 107-111 - also handled in common resource calling get

File: docs/resources/global_init_script.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+```
+
+You can also create managed global init script with inline sources through `content_base64` attribute.
+
+```hcl
+resource "databricks_global_init_script" "init2" {
+  content_base64 = base64encode(<<-EOT
+    #!/bin/bash
+    echo "hellp world"
```
Comment (nfx) [inline docs/resources/global_init_script.md:22]: image

File: docs/resources/global_init_script.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+-> **Note** Global init script in the Databricks workspace would only be changed, if Terraform stage did change. This means that any manual changes to managed global init script won't be overwritten by Terraform, if there's no local change to source.
+
+The size of a global init script source code must not exceed 64Kb. The following arguments are supported:
+
+* `name` (string, required) - the name of the script.  It should be unique
+* `source` - Path to script's source code on local filesystem. Conflicts with `content_base64`
+* `content_base64` - The base64-encoded source code global init script. Conflicts with `source`. Use of `content_base64` is discouraged, as it's increasing memory footprint of Terraform state and should only be used in exceptional circumstances
+* `enabled` (bool, optional default: `false`) if the script is enabled, or not
+* `position` (int, optional, default: 10000) the position of the script in the list of scripts.  With default value it's added at the end of the list.  The behavior of the REST API may lead to a situation that
```
Comment (nfx) [inline docs/resources/global_init_script.md:39]: didn't finish sentence :) see my previous comments on this attribute

**Review by nfx — CHANGES_REQUESTED** [review]
great work! comments:

1. i think it's more of `workspace` type of resource, not a compute.
2. Unit tests should be on resource CRUD, rather on API wrapper
3. `position` probably should be ignored and just returned as `computed`. e.g. not allowing to specify it.

File: compute/global_init_scripts.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	err := a.client.Get(a.context, "/global-init-scripts/"+scriptID, nil, &initScript)
+	return initScript, err
+}
+
+// Delete deletes specific global init scripts
+func (a GlobalInitScriptsAPI) Delete(scriptID string) error {
+	request := map[string]string{
+		"script_id": scriptID,
+	}
+	err := a.client.Delete(a.context, "/global-init-scripts/"+scriptID, request)
```
[the remark this answers is not in the document]
Comment (alexott) [reply compute/global_init_scripts.go:68]: that's a good question - per documentation we need path param only, but it doesn't work - raised the issue with docs team, but this requires investigation

**Comment by alexott** [comment]
@nfx comments are addressed

**Review by nfx — APPROVED** [review]

**Pull request merged**